### PR TITLE
FA-124 Fragment Create Account Implementation

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

**Comment by ForzenAcademy** [comment]
Rebase this

**Comment by HamdanSCode** [comment]
I feel like this could benefit from splitting the module holding the view models for legacy into two, one for fragments and one for normal legacy, reason being that the Navigator and FragmentNavigator needs to be provided to both and both the fragment and normal view model need to act slightly differently.

File: appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.hamdan.forzenbook.legacy.fragment.app.view
+
+import android.content.Context
+import com.hamdan.forzenbook.legacy.core.view.Navigator
+
+class NavigatorImpl:Navigator {
```
Comment (ForzenAcademy) [inline appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt]: Unformatted

File: appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    override fun navigateToLogin(context: Context) {
```
Comment (ForzenAcademy) [inline appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt:8]: TODO or finish this

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/view/FragmentNavigatorImpl.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.hamdan.forzenbook.applegacy.view
+
+import com.hamdan.forzenbook.legacy.core.view.FragmentNavigator
+
+class FragmentNavigatorImpl : FragmentNavigator {
+    override fun navigateToLogin() {
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/view/FragmentNavigatorImpl.kt:7]: Finish or comment

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt [hunk 4]
```diff
@@ -110,7 +115,8 @@ abstract class BaseCreateAccountViewModel(
                             birthDay = Entry(text = "", error = LoginError.BirthDateError.TooYoung),
                             email = Entry(text = "", error = LoginError.EmailError.Valid),
                             location = Entry(text = "", error = LoginError.LocationError.Length),
-                            isLoading = false
+                            isLoading = false,
+                            accountCreated = true
```
Comment (ForzenAcademy) [inline createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt]: Always leave trailing commas

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt [hunk 5]
```diff
@@ -151,7 +157,8 @@ fun BaseCreateAccountViewModel.CreateAccountState.toCreateAccountUiState(): Crea
         email = this.email,
         location = this.location,
         isDateDialogOpen = this.isDateDialogOpen,
-        isLoading = this.isLoading
+        isLoading = this.isLoading,
+        accountCreated = this.accountCreated
```
Comment (ForzenAcademy) [inline createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt]: Get in habit of trailing commas so you don't get diffs like this

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginFragmentViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        get() = _state
+
+    override var loginState: LoginState
+        get() = _state.value
+        set(value) {
+            _state.value = value
+        }
+
+    fun logIn() {
+        // use navigator to navigate to main page
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginFragmentViewModel.kt]: Ticket number

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginFragmentViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    val state: StateFlow<LoginState>
+        get() = _state
+
+    override var loginState: LoginState
+        get() = _state.value
+        set(value) {
+            _state.value = value
+        }
+
+    fun logIn() {
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginFragmentViewModel.kt]: `login`

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginViewModel.kt [hunk 8]
```diff
@@ -35,6 +35,10 @@ class LegacyLoginViewModel @Inject constructor(
             _state.value = value
         }
 
+    fun logIn(context: Context) {
```
Comment (ForzenAcademy) [inline legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/viewmodels/LegacyLoginViewModel.kt]: `login`

File: loginCompose/src/main/java/com/hamdan/forzenbook/login/compose/MainLogin.kt [hunk 9]
```diff
@@ -40,6 +42,13 @@ fun MainLoginContent(
     onSubmission: () -> Unit,
     onCreateAccountPress: () -> Unit,
 ) {
+    val navigator = LocalNavController.current
+    LaunchedEffect(state.loggedIn) {
+        if (state.loggedIn) {
+            // Todo when a main page is implemented create a navigation to it here
```
Comment (ForzenAcademy) [inline loginCompose/src/main/java/com/hamdan/forzenbook/login/compose/MainLogin.kt:48]: Ticket name

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt (hunk 5 above)
Comment (HamdanSCode) [reply createAccountCore/src/main/java/com/hamdan/forzenbook/createaccount/core/viewmodel/BaseCreateAccountViewModel.kt]: ye for sure, I'm a little inconsistent with it, but have been getting better about it though.

File: loginCompose/src/main/java/com/hamdan/forzenbook/login/compose/MainLogin.kt (hunk 9 above)
Comment (HamdanSCode) [reply loginCompose/src/main/java/com/hamdan/forzenbook/login/compose/MainLogin.kt:48]: theres no ticket for that yet.

File: legacyLoginFragment/src/main/java/com/hamdan/forzenbook/legacy/login/fragment/LoginFragment.kt [hunk 10]
```diff
@@ -78,7 +78,9 @@ class LoginFragment : Fragment() {
                 loginModel.state.collect { state ->
                     withContext(Dispatchers.Main) {
                         state.apply {
-
+                            if (this.loggedIn) {
+                                loginModel.login()
```
Comment (ForzenAcademy) [inline legacyLoginFragment/src/main/java/com/hamdan/forzenbook/legacy/login/fragment/LoginFragment.kt:82]: return after this? you wouldn't want to keep going in this function right?

**Review by ForzenAcademy — APPROVED** [review]
Do comment

File: legacyLoginFragment/src/main/java/com/hamdan/forzenbook/legacy/login/fragment/LoginFragment.kt (hunk 10 above)
Comment (HamdanSCode) [reply legacyLoginFragment/src/main/java/com/hamdan/forzenbook/legacy/login/fragment/LoginFragment.kt:82]: Yea thats fair.

**Pull request merged**